tests/test_preprocessing.py

```diff
@@ -217,5 +219,6 @@ def test_tpu_rotate(self):
             syms.extend(
                 map(lambda r: self.find_symmetry(*r), zip(x, pi, x2, pi2)))
 
+        difference = set(symmetries.SYMMETRIES) - set(syms)
```

Comment:
I'm not 100% sure I understand how this test works - could you explain?

Reply:
run_one, run_two, and run_three read 100 examples from a tf_examples file.
run_two and run_three are the same because I call reset_random() (and because the random_uniform has a seed) this is deterministic.
run_one should have no rotation.

I then use "find_symmetry" to find which sym makes `apply_sym(sym, run_two[i]) == run_one[i]`
This is nice because it uses a second, well tested, implementation to verify `symmetries._rotate_train()` code path 

This final line is verifying that some position in run_two was rotated (or technically rotated inversely) with each symmetries so that I know I'm doing all rotations (and not forgetting some symmetries or wrongly implementing them)